Makefile

```diff
@@ -149,6 +149,11 @@ ifndef LLAMA_NO_ACCELERATE
 	endif
 endif # LLAMA_NO_ACCELERATE
 
+ifdef LLAMA_MPI
+	CFLAGS += -DGGML_USE_MPI -Wno-cast-qual -Wno-int-to-void-pointer-cast -Wno-void-pointer-to-int-cast
+	CXXFLAGS += -DGGML_USE_MPI -Wno-cast-qual
```

Comment:
Curious, why is it necessary to disable these warnings? Or is it just for development?

Reply:
I should be able to get rid of them:

* The int-to-pointer stuff was needed as I was storing the `dst_rank` in the `extra` pointer. In https://github.com/ggerganov/ggml/pull/340, I've added a new struct field that makes this unnecessary.
* The `no-cast-qual` flag is needed to suppress warnings in the MPI headers (`<mpi.h>`). Maybe source-level pragmas can work around this?